### DISCUSSION TITLE
How do I properly call obs_shutdown() without resulting in a crash?

### DISCUSSION THREAD

**Discussion by HeedfulCrayon** [upgradeQ/Streaming-Software-Scripting-Reference#15] [Q&A]
I am writing a script to be able to control OBS via MQTT through Home Assistant.  I want to be able to gracefully close OBS Studio when an MQTT message is detected, but when I try to call `obs_shutdown()` I get a crash window 
![image](https://user-images.githubusercontent.com/5224972/116907195-7ea94f00-abfe-11eb-9e0d-decee4cf5dac.png)

The crash indicates that the python scripting dll is at fault.  Any idea of how I can work around that?
It seems like the `obs_shutdown()` does not initiate `script_unload()`, and if I try to manually call it with removing the `obs_frontend_remove_event_callback()` and the `timer_remove()` it crashes as well

**Comment by upgradeQ** [comment]
Did you try unload in `script_save`? It's looks like a perfect candidate to a bug report. Maybe it's fixed in v27

**Reply by HeedfulCrayon** [reply]
I didn't even override the `script_save` function.  Are you saying to call `script_unload` in that function?

**Reply by upgradeQ** [reply]
Yes, or try to move code from it to `script_save`
